### DISCUSSION TITLE
Login page not found

### DISCUSSION THREAD

**Discussion by deanh84** [firefly-iii/firefly-iii#6617] [Firefly III]
I've installed Firefly III in a TrueNAS Jail.  When accessing the site using just the IP address, I'm getting a 404 Not Found on the Login page redirect.  

I can include ./public/index.php in the URL, and the page will load, but without CSS, so I believe that the client session at least has access (htaccess) below ./public.  

The tech mix is:
- TrueNAS Core 2022
- FreeBSD 13.1 
- Apache 2.4
- PHP 8.0
- MySQL 8.0

Here is how document root is configured in the Apache **httpd.conf** file:
```
<Directory />
    AllowOverride none
    Require all denied
</Directory>

DocumentRoot "/usr/local/www/apache24/data/firefly-iii/public/"
<Directory "/usr/local/www/apache24/data/firefly-iii/public/">
    Options Indexes FollowSymLinks
    AllowOverride All
    Require all granted
</Directory>
```
All files under ./firefly-iii are owned by www:www, which is the user:group the Apache process is running with.

Here is the **.htaccess** file under ./firefly-iii/public/:
```
<IfModule mod_rewrite.c>
    <IfModule mod_negotiation.c>
        Options -MultiViews
    </IfModule>

    RewriteEngine On

    # Redirect Trailing Slashes If Not A Folder...
    RewriteCond %{REQUEST_FILENAME} !-d
    RewriteCond %{REQUEST_URI} (.+)/$
    RewriteRule ^ %1 [L,R=301]

    # Handle Front Controller...
    RewriteCond %{REQUEST_FILENAME} !-d
    RewriteCond %{REQUEST_FILENAME} !-f
    RewriteRule ^ index.php [L]

    # Handle Authorization Header
    RewriteCond %{HTTP:Authorization} .
    RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
</IfModule>
```
Any assistance in troubleshooting would be greatly appreciated!

**Comment by JC5** [comment]
Hey, there's your thread. Nice work. Question: does Apache have mod_rewrite enabled?

**Reply by deanh84** [reply]
Holy cow.  That was it!  I had to uncomment the following line in /usr/local/etc/apache24/httpd.conf:

```
LoadModule rewrite_module libexec/apache24/mod_rewrite.so
```
I'm learning more than I anticipated with this project....  Thanks for the help James!
